webapp/graphite/render/functions.py

```diff
@@ -1734,6 +1734,25 @@ def aliasSub(requestContext, seriesList, search, replace):
       series.name = re.sub(search, replace, series.name)
   return seriesList
 
+def aliasQuery(requestContext, seriesList, search, query, newName):
+  """
+  Performs a query to alias the metrics in seriesList.
+
+  .. code-block:: none
+
+    &target=aliasQuery(channel.power.*,"channel\.power\.([0-9]+)","channel.frequency.\\1", "Channel \\1 MHz")
+  """
+  for serie in seriesList:
+    newQuery = re.sub(search, query, serie.name)
+    newSeriesList = evaluateTarget(requestContext, newQuery)
+    if newSeriesList is None or len(newSeriesList) == 0:
+      raise Exception('No serie found with query: ' + newQuery)
```

Comment:
Also, I don't think this failure condition is tested in the tests provided.  Can you add another test or two to test the failure conditions?

The rest looks good!